drivers/display/display_gc9x01x.c

```diff
@@ -0,0 +1,745 @@
+/**
+ * Copyright (c) 2023 Mr Beam Lasers GmbH.
+ * Copyright (c) 2023 Amrith Venkat Kesavamoorthi <amrith@mr-beam.org>
+ * Copyright (c) 2023 Martin Kiepfer <mrmarteng@teleschirm.org>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "display_gc9x01x.h"
+
+#include <zephyr/dt-bindings/display/gc9x01x.h>
+#include <zephyr/drivers/pwm.h>
+#include <zephyr/drivers/display.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(display_gc9x01x, CONFIG_DISPLAY_LOG_LEVEL);
+
+/* Command/data GPIO level for commands. */
+#define GC9X01X_GPIO_LEVEL_CMD  0U
+/* Command/data GPIO level for data. */
+#define GC9X01X_GPIO_LEVEL_DATA 1U
+
+/* Maximum number of default init registers  */
+#define GC9X01X_NUM_DEFAULT_INIT_REGS (16U)
```

Comment:
According to commands listed below, the maximum number is 12. Would it be better to lower this value from 16 to 12?

Reply:
I left some buffer, but I agree that it is better to keep things tight!